interview-questions.md

```diff
@@ -6,46 +6,169 @@ First, without external resources. Challenge yourself to answer from memory.
+
+In summary, just like power-up items in Sonic the Hedgehog enhance Sonic's abilities, gems in Ruby provide pre-built functionalities that can be added to Ruby projects to extend their capabilities and make development tasks easier and more efficient.
+
+---
+
+### 3. What is a relational database? Are there other kinds of databases?
 
 Your answer:
 
+A relational database is like a spreadsheet with tables, where data is organized in rows and columns with defined relationships. Another type of database is a NoSQL database. I'm sure there are others but at this time I have no knowledge of what they are.
+
 Researched answer:
 
-4. What are primary keys? Why are they important?
+A relational database is like a city planning office in Cities: Skylines. It stores data in tables with rows and columns, where relationships between data elements are defined by keys. Just like how the city planning office organizes data in tables with different attributes for each building or infrastructure.
+
+On the other hand, other types of databases, such as flat-file, hierarchical, network, and NoSQL databases, are like different ways of organizing data in Cities: Skylines based on data models and structures. For example:
+
+Flat-file database: It's like a simple text file with data separated by delimiters, similar to how a list of building information may be stored in a text file in Cities: Skylines.
+
+Hierarchical database: It's like a tree-like structure in Cities: Skylines, where buildings and infrastructure are organized in a hierarchy with parent and child nodes, similar to how different types of buildings are organized under a specific category or zone.
+
+Network database: It's like a more complex tree-like structure in Cities: Skylines, where buildings and infrastructure can have multiple connections and relationships with other elements, similar to how different buildings may be interconnected with roads, power lines, or water pipes in the game.
+
+NoSQL database: It's like using different data models, such as key-value, document, column-family, or graph, to store and manage data in different ways in Cities: Skylines. For example, using a key-value model to store data on available resources or using a graph model to represent the relationships between different districts or neighborhoods in the city.
+
+In summary, a relational database is one way of organizing data, while other types of databases provide different approaches based on data models and structures, just like how different methods of organizing data are used in Cities: Skylines to manage the virtual city efficiently.
+
```

Comment:
Great examples! You must love Cities: Skylines.....

Reply:
I used to play it for hours at a time! 😅